triage failing examples

[examples/read/now.eg.ts]
```
vscode ➜ /workspaces/capi (feat/trun-workflows) $ deno task run examples/read/now.eg.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/read/now.eg.ts"
Task capi deno run -A main.ts "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/read/now.eg.ts"
Capi server listening at "http://localhost:4646/"
A new release of Deno is available: 1.31.1 → 1.32.3 Run `deno upgrade` to install it.
Release notes: https://github.com/denoland/deno/releases/tag/v1.32.3
Blog post: https://deno.com/blog/v1.32
Now: 1680534228000n
error: Uncaught (in promise) ScaleAssertError: typeof value !== "number"
      throw new ScaleAssertError(codec, this.value, `typeof ${this.path} !== "${type}"`)
            ^
    at AssertState.typeof (https://deno.land/x/scale@v0.11.2/common/assert.ts:25:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.2/codecs/int.ts:37:20)
    at Module.assert (https://deno.land/x/scale@v0.11.2/common/codec.ts:140:9)
    at file:///workspaces/capi/examples/read/now.eg.ts:15:3
inner command exited with status code 1
```

[examples/tx/handle_errors.eg.ts]
```
vscode ➜ /workspaces/capi (feat/trun-workflows) $ deno task run examples/tx/handle_errors.eg.ts
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/tx/handle_errors.eg.ts"
Task capi deno run -A main.ts "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/tx/handle_errors.eg.ts"
Capi server listening at "http://localhost:4646/"
Downloading polkadot/v0.9.39-1/x86_64-unknown-linux-gnu
Transfer: ready
Transfer: { inBlock: "0x6466fd5435ad61cce0a6c4379b53b57cf300270569a37a57d7251cbdc41fdc39" }
Transfer: { finalized: "0x6466fd5435ad61cce0a6c4379b53b57cf300270569a37a57d7251cbdc41fdc39" }
error: Uncaught (in promise) AssertionError: A non-Error object was rejected.
        throw new AssertionError(`A non-Error object was rejected${msgSuffix}`);
              ^
    at assertRejects (https://deno.land/std@0.182.0/testing/asserts.ts:888:15)
inner command exited with status code 1
```

[examples/tx/examples/watch.eg.ts]
```
vscode ➜ /workspaces/capi (feat/trun-workflows) $ deno task run examples/watch.eg.ts           
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/watch.eg.ts"
Task capi deno run -A main.ts "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/watch.eg.ts"
Capi server listening at "http://localhost:4646/"
1680534942001n
error: Uncaught (in promise) ScaleAssertError: typeof value !== "number"
      throw new ScaleAssertError(codec, this.value, `typeof ${this.path} !== "${type}"`)
            ^
    at AssertState.typeof (https://deno.land/x/scale@v0.11.2/common/assert.ts:25:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.2/codecs/int.ts:37:20)
    at Module.assert (https://deno.land/x/scale@v0.11.2/common/codec.ts:140:9)
    at file:///workspaces/capi/examples/watch.eg.ts:24:5
inner command exited with status code 1
```

[examples/paginate.eg.ts ]
```
vscode ➜ /workspaces/capi (feat/trun-workflows) $ deno task run examples/paginate.eg.ts 
Task run deno task capi -- deno run -A -r=http://localhost:4646/ "examples/paginate.eg.ts"
Task capi deno run -A main.ts "--" "deno" "run" "-A" "-r=http://localhost:4646/" "examples/paginate.eg.ts"
Capi server listening at "http://localhost:4646/"
Account keys: [
  Uint8Array(32) [
     36, 190, 125, 135, 166, 168,  96, 179,
     97, 246, 247, 132, 162, 141,  29,  35,
     26,  35, 122,  25,  51, 164, 233,   8,
    250,  92, 215, 169,  29,  22, 181,  45
  ],
  Uint8Array(32) [
    104,  97,  95, 169,  32, 143, 100, 148,
    188,  64, 222,  88, 241,   6,  45, 114,
     55, 156,   3, 213,  83, 197,   5,  30,
    147, 185,  22, 160,  79, 227,  54,  38
  ],
  Uint8Array(32) [
    152, 194,  98, 174,  34,  89,  83, 217,
     88,   1,   6,  99,  82, 233,  96, 132,
    203, 144, 139,  46, 178,  89, 168,  83,
    112,  63, 181,  73,  13, 128, 152,  38
  ],
  Uint8Array(32) [
    208,  88, 242, 203,  78, 155, 191, 217,
      6,  39,  31, 117,  95, 111,  90, 159,
     17, 236,  60, 134, 235, 253, 234,  62,
    104,  31, 206, 115, 119, 208, 196,  90
  ],
  Uint8Array(32) [
    166, 181,  39, 159, 172, 231, 147, 161,
     30, 230,  22, 119,  27, 161,  13, 241,
     99, 225, 165, 178, 122, 151,   7, 102,
    251,  57, 238,  34,  37, 176,  54, 115
  ],
  Uint8Array(32) [
      4,  42,  76,  70, 199,   2,  94, 153,
     65, 186,  95,  14, 169, 116, 190, 225,
    232, 155,  59, 247, 157, 187,  99,  51,
    181, 227, 130, 133, 244, 204,  36,  55
  ],
  Uint8Array(32) [
     66, 104, 213, 185, 239, 212,  38, 168,
    227,  97,  82, 105, 206,  21, 127, 201,
     16,  70, 243, 155,  16,  43, 205, 205,
     97, 154, 255,  46,  56, 204, 148, 122
  ],
  Uint8Array(32) [
    212, 221, 160, 166,  40,  17, 140, 196,
    216, 139, 213, 249, 152, 157, 100, 220,
      6, 176, 125, 167, 215, 252,  72,  16,
     94, 116, 132,  78,  25,  23,  36,  91
  ],
  Uint8Array(32) [
    106,  46, 129, 225, 18, 198, 175,  61,
    145,  94, 152, 254, 23, 115, 168,  56,
    185,  92, 204, 164,  8,  95, 168, 139,
    241, 160, 249, 215, 29,  77, 185, 106
  ],
  Uint8Array(32) [
    140, 238,   8,  74, 133,  93, 203, 141,
    217, 179, 207, 190, 210,  24, 170,  14,
      3, 107, 196, 178,  86, 172,  66, 175,
      5, 174, 157,  37, 209,  63,   4,  26
  ]
]
Collections entries: ValueRune { _prime: [Function] }
error: Uncaught (in promise) ScaleAssertError: !(value instanceof Array)
      throw new ScaleAssertError(codec, this.value, `!(${this.path} instanceof ${ctor.name})`)
            ^
    at AssertState.instanceof (https://deno.land/x/scale@v0.11.2/common/assert.ts:37:13)
    at Codec._assert (https://deno.land/x/scale@v0.11.2/codecs/array.ts:32:24)
    at Module.assert (https://deno.land/x/scale@v0.11.2/common/codec.ts:140:9)
    at file:///workspaces/capi/examples/paginate.eg.ts:24:3
inner command exited with status code 1
```

--- a/examples/.ignore
+++ b/examples/.ignore
@@ -1,4 +1,3 @@
-multisig/basic.ts
-multisig/stash.ts
-multisig/virtual.ts
+ink/deploy.eg.ts
+smoldot.eg.ts
 xcm/asset_teleportation.ts

--- a/examples/multisig/virtual.eg.ts
+++ b/examples/multisig/virtual.eg.ts
@@ -81,7 +81,7 @@ const call = Balances.transfer({
 })
 
 // Fund Billy and Carol's proxy accounts (existential deposits).
-await fundAndRatify("billy", carol).run()
+await fundAndRatify("billy", billy).run()
 await fundAndRatify("carol", carol).run()
 
 // Retrieve David's final balance.

--- a/examples/paginate.eg.ts
+++ b/examples/paginate.eg.ts
@@ -6,8 +6,7 @@
  */
 
 import { $ } from "capi"
-import { System } from "polkadot_dev/mod.js"
-import { types, Uniques } from "statemint/mod.js"
+import { System, types } from "polkadot_dev/mod.js"
 
 // Reference the first 10 keys of a polkadot dev chain's system account map.
 const accountKeys = await System.Account.keyPage(10, null).run()
@@ -16,12 +15,12 @@ const accountKeys = await System.Account.keyPage(10, null).run()
 console.log("Account keys:", accountKeys)
 $.assert($.sizedArray($.uint8Array, 10), accountKeys)
 
-// Reference the first 10 key-value pairs of a statemint dev chain's uniques class map.
-const collectionsEntries = await Uniques.Class.entryPage(10, null).run()
+// Reference the first 10 key-value pairs of a polkadot dev chain's system account map.
+const accountEntries = await System.Account.entryPage(10, null).run()
 
-// Each entry should be of type `[number, types.pallet_uniques.types.CollectionDetails]`
-console.log("Collections entries:", collectionsEntries)
+// Each entry should be of type `[Uint8Array, AccountInfo]`
+console.log("Account entries:", accountEntries)
 $.assert(
-  $.sizedArray($.tuple($.u32, types.pallet_uniques.types.$collectionDetails), 10),
-  collectionsEntries,
+  $.sizedArray($.tuple($.uint8Array, types.frame_system.$accountInfo), 10),
+  accountEntries,
 )

--- a/examples/paginate.eg.ts
+++ b/examples/paginate.eg.ts
@@ -22,6 +22,6 @@ const collectionsEntries = await Uniques.Class.entryPage(10, null).run()
 // Each entry should be of type `[number, types.pallet_uniques.types.CollectionDetails]`
 console.log("Collections entries:", collectionsEntries)
 $.assert(
-  $.sizedArray($.tuple($.u16, types.pallet_uniques.types.$collectionDetails), 10),
+  $.sizedArray($.tuple($.u32, types.pallet_uniques.types.$collectionDetails), 10),
   collectionsEntries,
 )

--- a/examples/paginate.eg.ts
+++ b/examples/paginate.eg.ts
@@ -17,11 +17,11 @@ console.log("Account keys:", accountKeys)
 $.assert($.sizedArray($.uint8Array, 10), accountKeys)
 
 // Reference the first 10 key-value pairs of a statemint dev chain's uniques class map.
-const collectionsEntries = Uniques.Class.entryPage(10, null)
+const collectionsEntries = await Uniques.Class.entryPage(10, null).run()
 
 // Each entry should be of type `[number, types.pallet_uniques.types.CollectionDetails]`
 console.log("Collections entries:", collectionsEntries)
 $.assert(
-  $.sizedArray($.tuple($.u8, types.pallet_uniques.types.$collectionDetails), 10),
+  $.sizedArray($.tuple($.u16, types.pallet_uniques.types.$collectionDetails), 10),
   collectionsEntries,
 )

--- a/examples/read/now.eg.ts
+++ b/examples/read/now.eg.ts
@@ -12,4 +12,4 @@ const now = await Timestamp.Now.value().run()
 console.log("Now:", now)
 
 // Ensure `now` is of the correct shape.
-$.assert($.u32, now)
+$.assert($.u64, now)

--- a/examples/sign/ed25519.eg.ts
+++ b/examples/sign/ed25519.eg.ts
@@ -8,8 +8,7 @@
 import { assert } from "asserts"
 import { Rune } from "capi"
 import { signature } from "capi/patterns/signature/polkadot.ts"
-import * as ed from "npm:@noble/ed25519"
-import { sha512 } from "npm:@noble/hashes/sha512"
+import * as ed from "https://esm.sh/@noble/ed25519@1.7.3"
 import { Balances, createUsers, System, types } from "westend_dev/mod.js"
 
 const { alexa, billy } = await createUsers()
@@ -22,23 +21,18 @@ const billyFree = System.Account
 const billyFreeInitial = await billyFree.run()
 console.log("Billy free initial:", billyFreeInitial)
 
-// Bring your own Ed25519 implementation. In this case, we utilize
-// [`noble-ed25519`](https://github.com/paulmillr/noble-ed25519), which
-// requires us to specify a sync sha512 hasher implementation.
-ed.etc.sha512Sync = (...msgs) => sha512(ed.etc.concatBytes(...msgs))
-
 // Initialize a secret with the `crypto.getRandomValues` builtin.
 const secret = crypto.getRandomValues(new Uint8Array(32))
 
 // Get a Rune of the secret-corresponding multiaddress.
 const address = types.sp_runtime.multiaddress.MultiAddress
-  .Id(ed.getPublicKey(secret))
+  .Id(await ed.getPublicKey(secret))
 
 // Define a `sign` function for later use.
-function sign(msg: Uint8Array) {
+async function sign(msg: Uint8Array) {
   return {
     type: "Ed25519" as const,
-    value: ed.sign(msg, secret),
+    value: await ed.sign(msg, secret),
   }
 }
 

--- a/examples/tx/handle_errors.eg.ts
+++ b/examples/tx/handle_errors.eg.ts
@@ -22,7 +22,7 @@ const extrinsicError = await Balances
   .dbgStatus("Transfer:")
   .inBlockEvents()
   .unhandleFailed()
-  .rehandle(ExtrinsicError, (error) => error)
+  .rehandle(ExtrinsicError)
   .run()
 
 // Ensure `extrinsicError` is in fact an instance of `ExtrinsicError`

--- a/examples/tx/handle_errors.eg.ts
+++ b/examples/tx/handle_errors.eg.ts
@@ -6,21 +6,25 @@
  * (and handling) of dispatch errors.
  */
 
-import { assertRejects } from "asserts"
+import { assertInstanceOf } from "asserts"
 import { alice, bob, ExtrinsicError } from "capi"
 import { signature } from "capi/patterns/signature/polkadot.ts"
-import { Balances } from "westend_dev/mod.js"
+import { Balances } from "contracts_dev/mod.js"
 
 // The following should reject with an `ExtrinsicError`.
-assertRejects(() =>
-  Balances
-    .transfer({
-      value: 1_000_000_000_000_000_000_000_000_000_000_000_000n,
-      dest: bob.address,
-    })
-    .signed(signature({ sender: alice }))
-    .sent()
-    .dbgStatus("Transfer:")
-    .finalizedEvents()
-    .unhandleFailed()
-    .run(), ExtrinsicError)
+const extrinsicError = await Balances
+  .transfer({
+    value: 1_000_000_000_000_000_000_000_000_000_000_000_000n,
+    dest: bob.address,
+  })
+  .signed(signature({ sender: alice }))
+  .sent()
+  .dbgStatus("Transfer:")
+  .inBlockEvents()
+  .unhandleFailed()
+  .rehandle(ExtrinsicError, (error) => error)
+  .run()
+
+// Ensure `extrinsicError` is in fact an instance of `ExtrinsicError`
+console.log("The unhandled extrinsic error:", extrinsicError)
+assertInstanceOf(extrinsicError, ExtrinsicError)

--- a/examples/watch.eg.ts
+++ b/examples/watch.eg.ts
@@ -21,6 +21,6 @@ let i = 0
 // gather and yield the `collection`-described data upon new blocks.
 for await (const item of now.iter()) {
   console.log(item)
-  $.assert($.u32, item)
+  $.assert($.u64, item)
   if (++i === 3) break
 }

